janitor/functions.py

```diff
@@ -1599,3 +1605,297 @@ def check(varname: str, value, expected_types: list):
                 varname=varname, expected_types=expected_types
             )
         )
+
+
+def _clean_accounting_column(x):
+    """
+    This function performs the logic for the `type == "accounting"`
+    attribute in make_currency_column_numeric.
+
+    It is intended to be used in a pandas `apply` method.
+    """
+
+    y = x.strip()
+    y = y.replace(",", "")
+    y = y.replace(")", "")
+    y = y.replace("(", "-")
+    if y == "-":
+        return 0.00
+    return float(y)
+
+
+def _make_currency_column_numeric(x, cast_non_numeric=None):
+    """
+    This function performs the logic for the changing cell values in
+    the make_currency_column_numeric fucntion.
+
+    It is intended to be used in a pandas `apply` method, after being passed
+    through `partial`.
+    """
+    acceptable_currency_characters = {
+        "-",
+        ".",
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "6",
+        "7",
+        "8",
+        "9",
+        "0",
+    }
+    if len(x) == 0:
+        return "ORIGINAL_NA"
+
+    if cast_non_numeric:
+        if x in cast_non_numeric.keys():
+            check(
+                "{%r: %r}" % (x, str(cast_non_numeric[x])),
+                cast_non_numeric[x],
+                [int, float],
+            )
+            return cast_non_numeric[x]
+        else:
+            return "".join(i for i in x if i in acceptable_currency_characters)
+    else:
+        return "".join(i for i in x if i in acceptable_currency_characters)
+
+
+def _replace_empty_string_with_none(x):
+    if isinstance(x, int):
+        return x
+
+    elif isinstance(x, float):
+        return x
+
+    elif len(x):
+        return x
+
+
+def _replace_original_empty_string_with_none(x):
+    if x != "ORIGINAL_NA":
+        return x
+
+
+@pf.register_dataframe_method
+def make_currency_column_numeric(
```

Comment:
@szuckerman pardon me for being nitpicky here, but I'm wondering if there's a more expressive verb than "make"? 

Java-land is littered with getters and setters because, and I'm wondering if we can avoid the same fate with makers.

In addition to that, since this is a currency-related function, does it make more sense to put this in the `finance.py` submodule?

Reply:
Sure; I'm not married to any of the names. 

What do you think of:
1. `convert_currency_column_to_num`?
2. `currency_column_to_numeric`?
3. `currency_column_to_num`?

Reply:
I went with `2`.